spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/prometheus/PrometheusMetricsExportAutoConfiguration.java

```diff
@@ -86,4 +97,109 @@ public PrometheusScrapeEndpoint prometheusEndpoint(
 
 	}
 
+	/**
+	 * Configuration for <a href="https://github.com/prometheus/pushgateway">Prometheus
+	 * Pushgateway</a>.
+	 *
+	 * @author David J. M. Karlsen
+	 */
+	@Configuration
+	@ConditionalOnClass(PushGateway.class)
+	@ConditionalOnProperty(prefix = "management.metrics.export.prometheus.pushgateway", name = "enabled", havingValue = "true", matchIfMissing = false)
```

Comment:
I think you can drop `havingValue` and `matchIfMissing` from this line. The defaults should suffice.